### PR TITLE
Update documentation for V5 support

### PR DESCRIPTION
**Pull request by JakeQuilty**
- README:
   - Removed old information
   - Added table of contents, required versions, contributing link
- CONTRIBUTING:
   - Updated to follow community standards
   - Added some testing and dev info
- CHANGELOG:
   - Added changes from https://github.com/cyberark/conjur-api-dotnet/pull/48
   - Added links to older releases

Linked to https://github.com/cyberark/conjur-api-dotnet/issues/60

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
+- [Requirements](#requirements)
+- [Building](#building)
+- [Methods](#methods)
+- [Example](#example)
+- [Contributing](#contributing)
+- [License](#license)
+
+## Requirements
+
+Ensure your Conjur/DAP version use the [V5 API](https://docs.conjur.org/latest/en/Content/Developer/lp_REST_API.htm?tocpath=Developer%7CREST%C2%A0APIs%7C_____0#rest-api).
```
Comment (izgeri) [inline README.md:18]:
```suggestion
- DAP v10+ or Conjur OSS v1+
```
Are there any requirements on the language version that they use?

File: README.md [hunk 2]
```diff
@@ -1,19 +1,30 @@
...
+
+For V4, use the [V4 branch](https://github.com/cyberark/conjur-api-dotnet/tree/v4)
```
Comment (izgeri) [inline README.md:20]:
```suggestion
For Conjur Enterprise V4, use the [V4 branch](https://github.com/cyberark/conjur-api-dotnet/tree/v4)
```

File: README.md (hunk 1 above)
Comment (JakeQuilty) [reply README.md:18]: I'll add these suggestions. I added the .NET language version requirement to [CONTRIBUTING](https://github.com/cyberark/conjur-api-dotnet/blob/community-repo-standards/CONTRIBUTING.md)

Comment (izgeri) [reply README.md]: but if I'm an end user, will this client only work with specific versions of .net?

Comment (JakeQuilty) [reply README.md]: In all honesty, I don't know. I only know that C#7 in the actual project breaks the test builds. I'm not sure for incorporating this into another project.

@uCatu do you know anything about this?

File: README.md [hunk 3]
```diff
@@ -1,19 +1,30 @@
 # Conjur API for .NET
 
-This is a *Draft* implementation of the .NET API for [V5 Conjur](https://developer.conjur.net/).
-This implementation includes an example that shows how to:
+Programmatic .NET access to the [Conjur API](https://docs.conjur.org/latest/en/Content/Developer/lp_REST_API.htm?tocpath=Developer%7CREST%C2%A0APIs%7C_____0#rest-api) (for both Conjur OSS and Enterprise/DAP versions).
```
Comment (sgnn7) [inline README.md:3]: Just change this to `[Conjur](https://conjur.org)` - I think that doc link you have here might be liable to break at a later date.

File: README.md [hunk 4]
```diff
@@ -111,46 +132,17 @@ Usage: Example  <applianceURL>
 
 `variableId`: The name of an existing variable in Conjur that has a value set and for which the `username` has execute permissions.
 
-`hostFactoryToken`: A host factory token. The easiest way to get a host factory token for testing is to add a hostfactory to a layer using the Conjur CLI command `conjur hostfactory create` and `conjur hostfactory token create`. Take the token returned from that call and pass it as the hostFactoryToken parameter to this example.
+`hostFactoryToken`: A host factory token. The easiest way to get a host factory token for testing is to add a hostfactory to a layer using 
+the Conjur CLI command `conjur hostfactory create` and `conjur hostfactory token create`. Take the token returned from that call 
+and pass it as the hostFactoryToken parameter to this example.
```
Comment (sgnn7) [inline README.md:137]: Word-wrap all your added lines to a 80-100 margin

File: README.md [hunk 5]
```diff
@@ -111,46 +132,17 @@ Usage: Example  <applianceURL>
...
-    //   parameters: privilege - string name of the priv to check for
-    bool isAllowed = conjurVariable.Check("execute");
-    if (!isAllowed)
-    {
-        Console.WriteLine("You do not have permissions to get the value of {0}", variableId);
-    }
-    else
-    {
-        Console.WriteLine("{0} has the value: {1}", variableId, conjurVariable.GetValue());
-    }
```
Comment (sgnn7) [inline README.md:151]: This example is still pretty nice (if it works with the new changes) though maybe it should be simplified to just get/set of a variable. I'd keep this and move it to the new location of `Examples` section at the very least though.

File: README.md [hunk 6]
```diff
@@ -111,46 +132,17 @@ Usage: Example  <applianceURL>
...
-```
+We welcome contributions of all kinds to this repository. For instructions on how to get started and descriptions
+of our development workflows, please see our [contributing guide](https://github.com/cyberark/conjur-api-dotnet/blob/master/CONTRIBUTING.md).
 
 ## License
 
 This repository is licensed under Apache License 2.0 - see [`LICENSE`](LICENSE) for more details.
+
+
```
Comment (sgnn7) [inline README.md:148]: Extra newlines not needed

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,9 +4,17 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
-## Unreleased
+## [Unreleased]
+### Removed
+- Conjur V4 API support
```
Comment (sgnn7) [inline CHANGELOG.md:9]: Removed section should probably be after added section

File: CHANGELOG.md [hunk 8]
```diff
@@ -4,9 +4,17 @@ All notable changes to this project will be documented in this file.
...
 
-## 1.4.0 - 2018-07-05
+### Added
+- Conjur V5 API support
+- Ability to control "limit" and "offset" for `ListUsers()` and `ListVariables()`
+- More README detail
+- More CONTRIBUTING detail
```
Comment (sgnn7) [inline CHANGELOG.md:15]: These last 2 bullet points about documentation are irrelevant for a changelog

File: README.md (hunk 5 above)
Comment (JakeQuilty) [reply README.md:151]: These examples are all still provided in the example app that comes bundled in the repo

Comment (JakeQuilty) [reply README.md:151]: `example/Program.cs Line 103`

File: README.md (hunk 1 above)
Comment (uCatu) [reply README.md]: @sashaCher 
Belive that there will be a minimum version of .net support from and above

**Comment by JakeQuilty** [comment]
Jenkins failure:

```
[2020-06-08T20:35:57.407Z] Status: Downloaded newer image for mono:latest
[2020-06-08T20:35:57.407Z]  ---> b074d513c5cd
[2020-06-08T20:35:57.407Z] Step 2/10 : ADD dividedmind.pub /tmp
[2020-06-08T20:36:00.690Z]  ---> 7dfa9be47098
[2020-06-08T20:36:00.690Z] Step 3/10 : RUN apt-key add - < /tmp/dividedmind.pub && rm /tmp/dividedmind.pub
[2020-06-08T20:36:00.690Z]  ---> Running in 853a4201b75c
[2020-06-08T20:36:00.690Z] E: gnupg, gnupg2 and gnupg1 do not seem to be installed, but one of them is required for this operation
[2020-06-08T20:36:00.951Z] The command '/bin/sh -c apt-key add - < /tmp/dividedmind.pub && rm /tmp/dividedmind.pub' returned a non-zero code: 255
[2020-06-08T20:36:00.951Z] Makefile:17: recipe for target 'rebuild' failed
[2020-06-08T20:36:00.951Z] make: *** [rebuild] Error 255
[2020-06-08T20:36:00.951Z] make: Leaving directory '/var/lib/jenkins/workspace/_dotnet_community-repo-standards/docker'
script returned exit code 2
```

This looks unrelated to this commit, because the commit before this passed and this last one only changed README.md and CHANGELOG.md

File: README.md (hunk 1 above)
Comment (JakeQuilty) [reply README.md]: Hey guys, I moved this to a separate issue(https://github.com/cyberark/conjur-api-dotnet/issues/62), so this PR could be merged in the meantime. Could we please move the conversation over there?

**Comment by JakeQuilty** [comment]
Ran the Jenkins tests again and it [PASSED](https://jenkins.conjur.net/blue/organizations/jenkins/cyberark--conjur-api-dotnet/detail/community-repo-standards/5/pipeline/). Merging

**Comment by JakeQuilty** [comment]
Moved the language version conversation above to a separate issue https://github.com/cyberark/conjur-api-dotnet/issues/62

File: README.md (hunk 1 above)
Comment (sashaCher) [reply README.md]: We should distinct between language support means with which toolset it's possible to compile this code (in our case the issue is that VS 2017 toolset needed in order to compile C#7 code) and between runtime support means what's minimal framework (mono, etc.) version that can run this code (it should be at least .net framework 4.5.2 according our own needs)

**Pull request merged**